modules/new_profile/php/profile.class.inc

```diff
@@ -120,8 +135,128 @@ class Profile extends \NDB_Page
                 ->withStatus(405)
                 ->withHeader("Allow", "GET,POST");
         }
+    }
 
+    /**
+     * Verifies the values from POST message.
+     *
+     * @param array $values the POST data received from frontend form.
+     *
+     * @return array the verified parsed POST response.
+     */
+    function processValidity(array $values): array
+    {
+        // Validate (dateTaken, dateTakenConfirm, sex) not set.
+        if (!isset($values['dateTaken'])
+            || !isset($values['dateTakenConfirm'])
+            || !isset($values['sex'])
+        ) {
+            return array('error' => 'Incomplete candidate fields.');
+        }
+        // Validate dob.
+        if (strlen($values['dateTaken']) > 2
+            && !$this->_checkDate($values['dateTakenConfirm'])
+        ) {
+            return array('error' => 'Birth date is not a valid date.');
+        } else if ($values['dateTaken'] != $values['dateTakenConfirm']) {
+            // Validate (dob1 & dob2) are not equal.
+            return array('error' => 'Birth date confirmation mismatch.');
+        }
+        $config =& \NDB_Config::singleton();
+        // Validate edc.
+        if ($config->getSetting('useEDC') == 'true'
+            && isset($values['edcDateTaken'])
+            && isset($values['edcDateTakenConfirm'])
+        ) {
+            if (strlen($values['edcDateTaken']) > 2
+                && !$this->_checkDate($values['edcDateTaken'])
+            ) {
+                return array('error' => 'EDC is not a valid date.');
+            } else if ($values['edcDateTaken'] != $values['edcDateTakenConfirm']) {
+                return array('error' => 'Estimated Due date fields must match.');
```

Comment:
this should say "Expected Date of Confinement fields must match."

EDC does not stand for "estimated due date"